### DISCUSSION TITLE
Welcome to guider Discussions!

### DISCUSSION THREAD

**Discussion by iipeace** [iipeace/guider#255] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by vnesic** [comment]
Hi @iipeace ,

Should the guider work out of the box when integrated into an android system app?
I wanted to measure cpu and network perfomance and  tried using guider with an app on an android 6 aosp build that I'm working on now.
It is integrated into the app with https://chaquo.com/chaquopy/.
It does not succeed in  generating the guider.out file, or to start recording, it reports that it cannot enable sched events. although i was able to do that from adb. 
It is integrated into a system app working in permissive mode.
Do you have any advice?

**Reply by iipeace** [reply]
Hi, @vnesic,
Do you want to use guider in android app?
Then you can not recording feature because it requires root permission and access permission to / dir.
How about top feature? 
```
guider.py top -o .
```

Can you use also ADB shell?

**Reply by vnesic** [reply]
Hi @iipeace ,

Yes,
 I want to use it in an android app. But because python cannot work by default I'm using https://chaquo.com/chaquopy/ to run python ( I don't know if that's relevant).
 I'm using trying to use "top -R 5 -o", but it doesn't generate the .out file.
 I have root permission, I'm running the  AOSP build in permissive mode, I've even set all the selinux permissions I have seen that are missing in the log.
 And yes, I can use adb shell.

**Reply by iipeace** [reply]
I used to execute Guider with prebuilt python on android shell.
You can refer to https://github.com/iipeace/portable/tree/master/python.
If your android machine was based on arm 32bit you can use the prebuilt binary in that repo.
Please refer to [link](https://github.com/kuri65536/python-for-android/issues/164) that I created for your aarch64 machine.

**Comment by jdson-fixstars** [comment]
Hi @iipeace, 

Do you have any plan for supporting QNX platform?

**Reply by iipeace** [reply]
A guy has ported the psutil package on QNX so it can be used in Guider, and he is currently preparing his contribution. 
A few months ago, I made some modifications to Guider to support QNX using that ported psutil package, and once his contribution is completed, Guider’s real-time monitoring features and some utility commands will be available on QNX.

**Reply by jdson-fixstars** [reply]
@iipeace thank you for the comments.
I understood Guider could be partially (not fully) supported on QNX platform with that guy's contribution (hopefully) in the future.

**Reply by iipeace** [reply]
@jdson-fixstars I'll notify you when the contribution is finished :)
